ProGetAutomation/Functions/Get-ProGetAsset.ps1

```diff
@@ -2,45 +2,62 @@ function Get-ProGetAsset
 {
     <#
         .SYNOPSIS
-        Gets assets from ProGet. 
+        Gets metadata about items in an asset directory.
 
         .DESCRIPTION
-        Get-ProGetAsset gets assets from ProGet. A session and Directory is required. 
-        An optional asset name may be added to get the content of the file. 
-        If no asset name is added the function will return a list of all files in the asset directory.
+        The `Get-ProGetAsset` function gets metadata from ProGet about assets. Pass the path to an asset directory to the `Directory` parameter. Information about all the files in that asset directory is returned. If the URL to an asset directory in ProGet is `https://proget.example.com/assets/versions/subdirectory/file`, the directory parameter is everything after `assets/` but before the filename `/file`, or `versions/subdirectory`.
 
-        .EXAMPLE
+        If you also pass a value to the `$Name` parameter, only files that match `$Name` in the directory will be returned. Wildcards are supported.
+
+        Pass a ProGet session object to the `$Session` parameter. This object controls what instance of ProGet to use and what credentials and/or API keys to use. Use the `New-ProGetSession` function to create session objects.
+
+        ##Examples
+
+        #Example 1
         Get-ProGetAsset -Session $session -Name 'myAsset' -Directory 'versions'
         
-        Returns contents of 'myAsset' if asset is found, otherwise returns 404.
+        Demonstrates how to get metadata about an asset. In this case, information about the `/assets/versions/myAsset` file is returned.
 
-        .Example
-        Get-ProGetAsset -Session $session -Directory 'versions'
+        #Example 2
+        Get-ProGetAsset -Session $session -Directory 'versions/subdirectory'
         
-        Returns list of files in the versions asset directory. If no files found an empty list is returned.
+        Demonstrates how to get metadata from all files in the `versions/subdirectory` asset directory. If no files found an empty list is returned.
 
     #>
     param(

```

Comment:
I played around with the ProGet asset directories, and let's change the `Get`, `Remove`, and `Set` functions to look like this.

```
param(
    $Session,
    $DirectoryName, # The name of the asset directory.
    $Path,  # The path to the asset. Everything after the `/dir/` segment required by the API.
    $Filter  # Replaces $Name parameter; only `Get` and `Remove` should have this.
)

# ...snip...

$uriPath = '/endpoints/{0}/dir/{1}' -f $DirectoryName,$Path
```